src/vivarium_public_health/results/mortality.py

```diff
@@ -90,28 +92,42 @@ def columns_required(self) -> List[str]:
 
     def setup(self, builder: Builder) -> None:
         self.clock = builder.time.clock()
-        self.causes_of_death = [
+        self.set_causes_of_death(builder)
+
+    def set_causes_of_death(self, builder: Builder) -> None:
+        causes_of_death = [
             cause
             for cause in builder.components.get_components_by_type(
                 tuple(self.mortality_classes)
             )
             if cause.has_excess_mortality
         ]
 
+        # Convert to SimpleCauses and add on other_causes and not_dead
```

Comment:
Why do we have to do this now?

Reply:
SimpleClass was added to easy in handling what is actually required for observers. An example is anemia in the maternal model, or "all_causes" for mortality. For anemia, we previously handled it by making a DiseaseState out of it and then setting it as a model, but that's hard to remember exactly what to do and also comes w/ a bunch of overhead for unused things (transitions, etc). All we _really_ need as far as observing the results of anemia are that it has state_id, model, and cause_type attributes. That's it. 

Probably more importantly, this class acts as a fail-quick device b/c if any given cause doesn't happen to have on of those attributes, it won't runtime error until the VERY end of the sim during results writing.